File: Assignment_41/assignment2.c
/////////////////////////////////////////////////////////////////////////////////////////////
//
//    File name    : assignment2.c
//    Description  : Write a program to print below pattern using recursion.
//    Author name  : Aryan Shailendrasingh Pardeshi
//    Date         : 19/07/2025
//
/////////////////////////////////////////////////////////////////////////////////////////////

// INPUT  5
// OUTPUT 1   2   3   4   5

#include<stdio.h>

void Display(int iNo)
{
    if(iNo > 0)
    {    
        Display(iNo - 1); 

        printf("%d   ",iNo);
    }
}

int main()
{
    int iValue = 0;

    printf("Enter number : ");
    scanf("%d",&iValue);

    Display(iValue);

    printf("\n");

    return 0;
}

File: Assignment_48/assignment2.java
//////////////////////////////////////////////////////////////////////////
// 
//    File name    :- assignment2.java
//    Description  :- Write a program which accepts a string from user 
//                    and display below pattern.
//    Author name  :- Aryan Shailendrasingh Pardeshi
//    Date         :- 20/08/2025
//
//////////////////////////////////////////////////////////////////////////

// INPUT:  Hello

/* OUTPUT:
            H   e   l   l   o
            H   e   l   l   
            H   e   l   
            H   e
            H  
*/

import java.util.*;

class Pattern
{
    public void Pattern(String str)
    {
        int length = str.length();

        for(int i = length; i >= 1; i--)   
        {
            for(int j = 0; j < i; j++) 
            {
                System.out.print(str.charAt(j) + "\t");
            }
            System.out.println();
        }
    }
}

class assignment2
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.print("Enter a string : ");
        String iValue = sobj.nextLine();

        Pattern pobj = new Pattern();
        pobj.Pattern(iValue);
    }
}
